src/app/modules/ix-forms/components/ix-dynamic-form/ix-dynamic-form-item/ix-dynamic-form-item.component.spec.ts

```diff
@@ -1,89 +1,254 @@
+  title: 'Label Input Int',
+  tooltip: undefined,
+  type: 'input',
+} as DynamicFormSchemaInput;
+
+const selectSchema = {
+  controlName: 'select',
+  editable: undefined,
+  hideEmpty: true,
+  required: true,
+  title: 'Label Select',
+  tooltip: undefined,
+  options: of([]),
+  type: 'select',
+} as DynamicFormSchemaSelect;
+
+const checkboxSchema = {
+  controlName: 'checkbox',
+  editable: undefined,
+  indent: true,
+  required: undefined,
+  title: 'Label Checkbox',
+  tooltip: undefined,
+  type: 'checkbox',
+} as DynamicFormSchemaCheckbox;
+
+const ipaddrSchema = {
+  controlName: 'ipaddr',
+  editable: undefined,
+  required: undefined,
+  title: 'Static IP',
+  tooltip: undefined,
+  type: 'ipaddr',
+} as DynamicFormSchemaIpaddr;
+
+const explorerSchema = {
+  controlName: 'explorer',
+  type: 'explorer',
+} as DynamicFormSchemaExplorer;
+
+const listSchema = {
+  controlName: 'list',
   type: 'list',
-  title: 'DNS Options',
-  items: [
-    {
-      controlName: 'name',
-      type: 'input',
-      title: 'Option Name',
-      required: true,
-    },
-    {
-      controlName: 'value',
-      type: 'input',
-      title: 'Option Value',
-      required: true,
-    },
-  ],
-  itemsSchema: [
-    {
-      variable: 'name',
-      label: 'Option Name',
-      schema: {
-        type: 'string',
-        required: true,
-      },
-    },
-    {
-      variable: 'value',
-      label: 'Option Value',
-      schema: {
-        type: 'string',
-        required: true,
-      },
-    },
+} as DynamicFormSchemaList;
+
+const dictSchema = {
+  controlName: 'dict',
+  title: 'Label Dict',
+  hidden: false,
+  attrs: [
+    inputSchema,
+    selectSchema,
+    checkboxSchema,
   ],
-} as DynamicFormSchemaNode;
+  type: 'dict',
+} as DynamicFormSchemaDict;
 
 describe('IxDynamicFormItemComponent', () => {
   let spectator: Spectator<IxDynamicFormItemComponent>;
   const createComponent = createComponentFactory({
     component: IxDynamicFormItemComponent,
+    imports: [
+      FormsModule,
+      ReactiveFormsModule,
+      FlexLayoutModule,
+    ],
+    declarations: [
+      MockComponent(IxInputComponent),
+      MockComponent(IxListComponent),
+      MockComponent(IxSelectComponent),
+      MockComponent(IxCheckboxComponent),
+      MockComponent(IxIpInputWithNetmaskComponent),
+      MockComponent(IxExplorerComponent),
+    ],
   });
 
   beforeEach(() => {
     spectator = createComponent();
   });
 
-  it('isHidden', () => {
-    spectator.component.dynamicForm = dynamicForm;
-    spectator.component.dynamicSchema = dynamicSchema;
-    expect(spectator.component.isHidden).toBeTruthy();
-  });
+  describe('UI tests', () => {
+    it('input', () => {
+      spectator.component.dynamicForm = dynamicForm;
+      spectator.component.dynamicSchema = inputSchema;
+      spectator.detectComponentChanges();
+      expect(spectator.query('ix-input')).toBeVisible();
+      expect(spectator.query('ix-input')).not.toBeDisabled();
```

Comment:
I don't think it's every going to be disabled, as this probably just checks for [disabled] attribute on `ix-input` tag.
You can test for inputs being passed correctly:
```
expect(spectator.query(IxInputComponent).required).toBe(true);
```
etc.